pkg/graph/resolvers/status_evaluation_strategies.go

```diff
@@ -0,0 +1,143 @@
+package resolvers
+
+import (
+	"time"
+
+	"github.com/cmsgov/mint-app/pkg/shared/utilitytime"
+
+	"github.com/cmsgov/mint-app/pkg/models"
+)
+
+// GetAllStatusEvaluationStrategies returns all status evaluation strategies
+func GetAllStatusEvaluationStrategies() []StatusEvaluationStrategy {
+	return []StatusEvaluationStrategy{
+		&EndedStrategy{},
+		&ActiveStrategy{},
+		&AnnounceStrategy{},
+		&ClearanceEndStrategy{},
+		&ClearanceStartStrategy{},
+		&ICIPCompleteStrategy{},
+	}
+}
+
+// StatusEvaluationStrategy is an interface for evaluating the status of a model plan
+type StatusEvaluationStrategy interface {
+	Evaluate(
+		modelPlanStatus models.ModelStatus,
+		planBasics *models.PlanBasics,
+	) ([]models.ModelStatus, bool)
+}
+
+// ICIPCompleteStrategy is a strategy for evaluating the ICIP complete status of a model plan
+type ICIPCompleteStrategy struct{}
+
+// Evaluate returns the models.ModelStatusIcipComplete status if the model plan
+// status is not "ICIP complete" and the ICIP complete date has passed
+func (s *ICIPCompleteStrategy) Evaluate(
+	modelPlanStatus models.ModelStatus,
+	planBasics *models.PlanBasics,
+) ([]models.ModelStatus, bool) {
+	if models.GetModelStatusChronologicalIndex(modelPlanStatus) <
+		models.GetModelStatusChronologicalIndex(models.ModelStatusIcipComplete) &&
+		!utilitytime.IsTimeNilOrZero(planBasics.CompleteICIP) &&
+		planBasics.CompleteICIP.Before(time.Now()) {
+		return []models.ModelStatus{models.ModelStatusIcipComplete}, true
+	}
+	return nil, false
+}
+
+// ClearanceStartStrategy is a strategy for evaluating the clearance start status of a model plan
+type ClearanceStartStrategy struct{}
+
+// Evaluate returns the chronologically earliest clearance status
+// (models.ModelStatusInternalCmmiClearance) if the model plan status is not
+// "Plan Complete" and the clearance start date has passed
+func (s *ClearanceStartStrategy) Evaluate(
+	modelPlanStatus models.ModelStatus,
+	planBasics *models.PlanBasics,
+) ([]models.ModelStatus, bool) {
+	if models.GetModelStatusChronologicalIndex(modelPlanStatus) <
+		models.GetModelStatusChronologicalIndex(models.ModelStatusInternalCmmiClearance) &&
```

Comment:
I might just wanna double check this one against the AC, since it's one that is a bit more complex than the others:
> “Clearance start date” date in anticipated timeline has passed but status is not any of the clearance statuses or a future status

The `but status is not any of the clearance statuses` part makes me think we should check the _latest_ (chronologically) clearance status, since `ModelStatusInternalCmmiClearance: 3`, but a`ModelStatusOmbAsrfClearance: 6`

I only suggest it because right now if your `ClearanceStarts` date has passed, but you're in `ModelStatusCmsClearance`, it'll still suggest an update (even though you're already in a clearance status)

Maybe worth another chat with Natalie to see if we're interpreting the A.C. correctly on that one!

Reply:
Upon second review (thanks @OddTomBrooks for the heads up!) I think this is behaving as intended as-is!

To clarify: We only want to suggest the 4 clearance statuses if you're not in ANY clearance status, and this statement (`models.GetModelStatusChronologicalIndex(modelPlanStatus) < models.GetModelStatusChronologicalIndex(models.ModelStatusInternalCmmiClearance`) will do just that!

Reply:
Since `ModelStatusInternalCmmiClearance` is the lowest clearance status, chronologically speaking.